src/directives/AddrResolver.js

```diff
@@ -34,157 +30,65 @@ const AddrResolver = {
-            el.parentNode.parentNode.appendChild(errorPar);
-          } else {
-            getMultiCoinAddress(ens, normalise(e), parentCurrency)
-              .then(address => {
-                if (!checkDarklist(address)) {
-                  removeElements();
-                  _this.hexAddress =
-                    parentCurrency === 'ETH'
-                      ? toChecksumAddress(address)
-                      : address;
-                  _this.isValidAddress = true;
-                  errorPar.innerText = address;
-                  vnode.elm.parentNode.parentNode.appendChild(errorPar);
-                }
-              })
-              .catch(() => {
-                if (parentCurrency === 'ETH') {
-                  ens
-                    .resolver(normalise(e))
-                    .addr()
-                    .then(address => {
-                      if (!checkDarklist(address)) {
-                        removeElements();
-                        _this.hexAddress = toChecksumAddress(address);
-                        _this.isValidAddress = true;
-                        errorPar.innerText = address;
-                        vnode.elm.parentNode.parentNode.appendChild(errorPar);
-                      }
-                    })
-                    .catch(() => {
-                      removeElements();
-                      // eslint-disable-next-line
-                    errorPar.innerText = _this.$t('ens.ens-resolver.network-not-found', { network: network.type.name[0]});
-                      _this.isValidAddress = false;
-                      _this.hexAddress = '';
-                      vnode.elm.parentNode.parentNode.appendChild(errorPar);
-                    });
-                } else {
-                  removeElements();
-                  // eslint-disable-next-line
-                    errorPar.innerText = _this.$t('ens.ens-resolver.network-not-found', { network: network.type.name[0]});
-                  _this.isValidAddress = false;
-                  _this.hexAddress = '';
-                  vnode.elm.parentNode.parentNode.appendChild(errorPar);
-                }
-              });
-          }
-        }
-      } else if (e !== '') {
-        const isValid = WAValidator.validate(e, parentCurrency);
-        if (isValid) {
-          _this.isValidAddress = isValid;
-          _this.hexAddress =
-            parentCurrency === 'ETH' ? toChecksumAddress(e) : e;
-        } else {
-          if (canValidate(parentCurrency)) {
-            const isValid = MAValidator.validate(e, parentCurrency);
-            if (isValid) {
-              _this.isValidAddress = isValid;
-              _this.hexAddress = e;
-            } else {
-              _this.isValidAddress = false;
-              _this.hexAddress = '';
+      try {
+        const address = await resolution.addressOrThrow(domain, parentCurrency);
+        if (address) {
```

Comment:
Might also want to add an error similar to what you did on [here](https://github.com/MyEtherWallet/MyEtherWallet/pull/1809/files#diff-25ab0cdc1d5941225b37bcab2160e95dR89) in the instance that it returns a blank address which I'm assuming it shouldn't